src/main/java/seedu/address/model/Transcript.java

```diff
@@ -94,6 +95,76 @@ public void removeModule(Module key) {
         modules.remove(key);
     }
 
+    /**
+     * Return the current CAP
+     * @return current cap score
+     */
+    public double getCurrentCap() {
+        return calculateCapScore();
+    }
+
+    /**
+     * Calculate CAP Score based on modules with scores
+     * @return cap: cap score
+     */
+    private double calculateCapScore() {
+        UniqueModuleList gradedModulesList = getGradedModulesList();
+        Iterator<Module> gradedModulesIterator = gradedModulesList.iterator();
+
+        double totalCap = 0;
+        double point;
+        int totalModuleCredit = 0;
+        int moduleCredit;
+        while (gradedModulesIterator.hasNext()) {
+            Module module = gradedModulesIterator.next();
+            moduleCredit = module.getCredits().value;
+            point = module.getGrade().getPoint();
+            totalCap += moduleCredit * point;
+            totalModuleCredit += moduleCredit;
+        }
+
+        double cap = 0;
+        if (totalModuleCredit > 0) {
+            cap = totalCap / totalModuleCredit;
+        }
+
+        return cap;
+    }
+
+    /**
+     * Filters for modules that is to be used for CAP calculation
+     * @return gradedModulesList: a list of modules used for CAP calculation
+     */
+    private UniqueModuleList getGradedModulesList() {
```

Comment:
Consider adding this function in UniqueModuleList instead:
```
public List<Module> getGradedModulesList() {
  return internalList.filtered(Module::hasCompleted).filtered(x -> x.getGrade().affectsCap());
}
```

Reply:
If that the case, i think you might want to consider adding `getFilteredList(Predicate)` which should be allowed since `UniqueModuleList` while making sure the list is unique, should also provide a efficient way of providing access to the list. I will leave it to you to decide.